c.c

```diff
@@ -1388,10 +1388,16 @@ static void skipToMatch (const char *const pair)
 	{
 		if (CollectingSignature)
 			vStringPut (Signature, c);
-		
 
 		if (c == begin)
 		{
+			char s = cppGetc();
+			if (s == '<' && s == c) {
+				// templates can still contain "<<" ot "<<="
+				break;
```

Comment:
There's still cases that it will miss - for example:

```
template <int P> class c {};
c< 8 > bVar;
c< 1<<8 > aVar;

template<int X> func( c< 1<<X> & aParam) {
}
```

it will fail to parse out the definition for func, only dumping out aVar/bVar - because it's punted on trying to keep things balanced, it's just going to walk forward until the {.